samples/subsys/nvs/src/main.c

```diff
@@ -68,25 +68,31 @@ void main(void)
 	uint8_t key[8], longarray[128];
 	uint32_t reboot_counter = 0U, reboot_counter_his;
 	struct flash_pages_info info;
+	const struct device *flash_dev;
 
 	/* define the nvs file system by settings with:
 	 *	sector_size equal to the pagesize,
 	 *	3 sectors
 	 *	starting at FLASH_AREA_OFFSET(storage)
 	 */
+	flash_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_flash_controller));
```

Comment:
Shouldn't it be rather
`DEVICE_DT_GET(DT_MTD_FROM_FIXED_PARTITION(DT_NODE_BY_FIXED_PARTITION_LABEL(storage)))`
this way this would be really the device that holds the `storage` partition.
Eh we are missing define for that.

Reply:
>  Eh we are missing define for that.

Are you saying to leave this as-is? My goal with this patch is just to clean up the DT access, not to change semantics. This is currently using `device_get_binding(DT_CHOSEN_ZEPHYR_FLASH_CONTROLLER_LABEL)`, so this line is returning the exact same device.

Reply:
I think that this is bug, because the `zephyr_flash_controller` does not have to correspond to the flash controller the `storage` partition is under. So okay maybe replace it as you did and log an issue for investigation.

Reply:
OK, I can change it, let's see if @Laczen confirms

Reply:
I can confirm this is a bug.

Reply:
Thanks @Laczen . I added a patch on top of the series that fixes the bug in the way @de-nordic suggested.